pkg/drivers/powerflex.go

```diff
@@ -70,96 +70,79 @@ func PrecheckPowerFlex(ctx context.Context, cr *csmv1.ContainerStorageModule, op
 		return fmt.Errorf("%s %s not supported", csmv1.PowerFlexName, cr.Spec.Driver.ConfigVersion)
 	}
 
-	mdmVar, err := GetMDMFromSecret(ctx, cr, ct)
+	// Check if MDM is set in the secret
+	_, err := GetMDMFromSecret(ctx, cr, ct)
 	if err != nil {
 		return err
 	}
-	var newmdm corev1.EnvVar
+
+	return nil
+}
+
+func SetSDCinitContainers(ctx context.Context, cr csmv1.ContainerStorageModule, ct client.Client) (csmv1.ContainerStorageModule, error) {
+	mdmVar, _ := GetMDMFromSecret(ctx, &cr, ct)
+
+	// Check if SDC is enabled
 	sdcEnabled := true
-	for _, env := range cr.Spec.Driver.Node.Envs {
-		if env.Name == "X_CSI_SDC_ENABLED" && env.Value == "false" {
-			sdcEnabled = false
+	if cr.Spec.Driver.Node != nil {
+		for _, env := range cr.Spec.Driver.Node.Envs {
+			if env.Name == "X_CSI_SDC_ENABLED" && env.Value == "false" {
+				sdcEnabled = false
+				break
+			}
 		}
 	}
 
-	newInitContainers := make([]csmv1.ContainerTemplate, 0)
+	// Update init containers
+	var newInitContainers []csmv1.ContainerTemplate
 	for _, initcontainer := range cr.Spec.Driver.InitContainers {
-		if initcontainer.Name != "sdc" {
-			newInitContainers = append(newInitContainers, initcontainer)
-		} else if initcontainer.Name == "sdc" && sdcEnabled {
-			k := 0
-			initenv := initcontainer.Envs
-			for c, env := range initenv {
+		if initcontainer.Name == "sdc" && sdcEnabled {
+			// Update MDM env variable for "sdc" container
+			for i, env := range initcontainer.Envs {
 				if env.Name == "MDM" {
-					env.Value = mdmVar
-					newmdm = env
-					k = c
+					initcontainer.Envs[i].Value = mdmVar
 					break
```

Comment:
@rishabhatdell Can you please check the scenario when no MDM(or no Envs are passed) value is passed from the CR in initContainers , it will update the MDM value by picking it from secrets.  Following is the scenario:

    initContainers:
      - image: docker.io/dellemc/sdc:4.5.2.1
        imagePullPolicy: IfNotPresent
        name: sdc
        envs:

Reply:
There is a defect for minimal samples on the same scenario: 30620. FYR

Reply:
I have added a check to ensure that if the MDM variable is empty, it will be populated from the secret for both the sdc and sdc-monitor containers. This update also helps resolve defect 30620.
https://github.com/dell/csm-operator/blob/01d2c78ad8e4a09e4be3f7a863732cd2b6f69530/pkg/drivers/powerflex.go#L141-L147